### PR TITLE
V22F-890 Calling SDASpeedDreams

### PR DESCRIPTION
**Pull request by KungFuDonkey**
Calls the function of speed dreams to pre simulate data

### CODE REVIEW

File: SDALib-CPP/SDALib-CPP/SDAUtils.hpp [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#pragma once
+#include "msgpack.hpp"
+#include <vector>
+#include <string>
+#include "sdalib_export.h"
+
+/// @brief				      Copies a msgpack buffer
+/// @param  p_sbuffer		  The msgpack buffer
+/// @param  p_destination     The destination buffer
+///	@param  p_destinationSize The size of the destination buffer
```
Comment (larsdekwant) [inline SDALib-CPP/SDALib-CPP/SDAUtils.hpp:10]: Remove tabs

File: SDALib-CPP/SDALib-CPP/CMakeLists.txt [hunk 2]
```diff
@@ -6,9 +6,11 @@ SET(SDALIB_HEADERS
 	SDADriver.hpp
 	SDAAction.hpp
 	SDAData.hpp
-	SDASpeedDreams.hpp
+	SDAUtils.hpp
```
Comment (larsdekwant) [inline SDALib-CPP/SDALib-CPP/CMakeLists.txt:9]: replace tabs in this file

File: SDALib-CPP/SDALib-CPP/SDA_portability.h [hunk 3]
```diff
@@ -7,6 +7,8 @@
 #pragma once
 
 #include "ipclib_portability.h"
+#define _SILENCE_EXPERIMENTAL_FILESYSTEM_DEPRECATION_WARNING 1
+#include <experimental/filesystem>
```
Comment (larsdekwant) [inline SDALib-CPP/SDALib-CPP/SDA_portability.h:11]: add `namespace filesystem = std::experimental::filesystem;`

Also these same defines are currently in `SDATypesConverter.h`, from which they can be removed

File: SDALib-CPP/SDALib-CPP/SDASpeedDreams.h [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+private:
+    tCar CarConstructor(const tCar &p_other, tCarElt *p_carElt);
+    tCar SimCarTable[1];
+    tdble SimDeltaTime;
+    tdble simSkidFactor[6]{0.4f, 0.3f, 0.2f, 0.1f, 0.0f, 0.0f};
+    tdble simDammageFactor[6]{0.0f, 0.1f, 0.3f, 0.5f, 0.8f, 1.0f};
+    void SimUpdate(tSituation *s, double deltaTime);
+    tdble SimTimeOfDay = 0;
+    double Tair = 273.15;
+    ;
```
Comment (suppergerrie2) [inline SDALib-CPP/SDALib-CPP/SDASpeedDreams.h:20]: Extra `;`?

**Review by larsdekwant — CHANGES_REQUESTED** [review]
Also pls check whether IPCLib is up-to-date, since I notice that all the copyright notices are getting removed again.

File: SDALib-CPP/SDALib-Python/SDATypesConverter.h [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+#include <Python.h>
+#endif
+#include <stdlib.h>
+#include <iostream>
+#define _SILENCE_EXPERIMENTAL_FILESYSTEM_DEPRECATION_WARNING 1
+#include <experimental/filesystem>
+
+namespace filesystem = std::experimental::filesystem;
+
+///@broef class for converting structs from Python to Cpp and back
```
Comment (suppergerrie2) [inline SDALib-CPP/SDALib-Python/SDATypesConverter.h:18]:
```suggestion
///@brief class for converting structs from Python to Cpp and back
```

**Review by suppergerrie2 — CHANGES_REQUESTED** [review]

**Review by larsdekwant — APPROVED** [review]
some pointer types are missing, but normal struct values are usable.

**Pull request merged**